### PR TITLE
IsEveryoneHere feature Embed

### PR DESCRIPTION
**Pull request by MabroukTBN**
- Featue now displays infomation using an Embed
- Updated _IsEveryoneHere_ to also display approved leaves
- Does not display reason for leave to non-admins

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -83,19 +83,18 @@ async def IsMemberWorking(ctx, discorduser, date = datetime.today().strftime('%d
 async def IsEveryoneHere(ctx):
 
     guild = client.guilds[0]
-    voice_channel = client.get_channel(int(os.getenv("MeetingChannel_id")))
-    fulltime_role = discord.utils.get(guild.roles, name = "Full Time")
-    fulltime_member_list = list(filter(lambda user: fulltime_role in user.roles, guild.members))
-    fulltime_member_list_in_voicechannel = list(filter(lambda user : fulltime_role in user.roles, voice_channel.members))
+    meeting_channel = client.get_channel(int(os.getenv("MeetingChannel_id")))
+    isAdmin = True if Utilities.IsAdmin(ctx.author) else False
```
Comment (AntWong-TheBadNomads) [inline main.py:87]:
```suggestion
    isAdmin = Utilities.IsAdmin(ctx.author)
```

True if true, false if false? Redundant. Probably won't then need the variable at all here as well.

File: Leave/leave_interface.py [hunk 2]
```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
...
+    role_member_list = list(filter(lambda member: role in member.roles, members_all))
+    role_member_in_voicechannel_list = list(filter(lambda member : role in member.roles, voice_channel_members))
+    not_here_all = list(set(role_member_list) - set(role_member_in_voicechannel_list))
+
+    approved_leaves = list(filter(lambda member : (IsMemberOnLeave(member.id, date))[0], not_here_all))
+    approved_leaves_names = [member.display_name for member in approved_leaves]
+    approved_leaves_reasons = [IsMemberOnLeave(member.id, date)[1] for member in approved_leaves]
+
+    unapproved_leaves = list(filter(lambda member : not (IsMemberOnLeave(member.id, date)[0]), not_here_all))
+    unapproved_leaves = [member.display_name for member in unapproved_leaves]
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_interface.py:170]: Calling this unapproved leaves is a little premature. This variable represents missing people, not people who are on an unapproved leave.

File: Leave/leave_interface.py [hunk 3]
```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
...
+
+    date = datetime.datetime.strptime(datetime.datetime.today().strftime('%d/%m/%Y'), '%d/%m/%Y')
+
+    role_member_list = list(filter(lambda member: role in member.roles, members_all))
+    role_member_in_voicechannel_list = list(filter(lambda member : role in member.roles, voice_channel_members))
+    not_here_all = list(set(role_member_list) - set(role_member_in_voicechannel_list))
+
+    approved_leaves = list(filter(lambda member : (IsMemberOnLeave(member.id, date))[0], not_here_all))
+    approved_leaves_names = [member.display_name for member in approved_leaves]
+    approved_leaves_reasons = [IsMemberOnLeave(member.id, date)[1] for member in approved_leaves]
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_interface.py:167]: Very messy. Avoid multiple arrays whose indexes share a relationship. When you find that pattern, instead go for a more extensive data structure, like a dictionary here instead.

File: Leave/leave_interface.py [hunk 4]
```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
 
     return leaves_array[0]["reason"]
 
+def GetMissingMembersByRole(members_all, role, voice_channel_members):
+
+    date = datetime.datetime.strptime(datetime.datetime.today().strftime('%d/%m/%Y'), '%d/%m/%Y')
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_interface.py:159]: What are you trying to do here?

File: Leave/leave_interface.py [hunk 5]
```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
...
+
+    role_member_list = list(filter(lambda member: role in member.roles, members_all))
+    role_member_in_voicechannel_list = list(filter(lambda member : role in member.roles, voice_channel_members))
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_interface.py:162]:
```suggestion
    role_members_in_voicechannel = list(filter(lambda member : role in member.roles, voice_channel_members))
```
Plurality + the word `list` is redundant. We want descriptive names while avoiding redundancy.

File: Leave/leave_interface.py [hunk 6]
```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
...
+    not_here_all = list(set(role_member_list) - set(role_member_in_voicechannel_list))
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_interface.py:163]: What do you mean by `all`?

File: UI.py [hunk 7]
```diff
@@ -208,6 +208,27 @@ def CreateIsMemberWorkingOptions():
     ]
 
     return member_options
+
+def CreateIsEveryoneHereEmbed(approved_list_names, approved_list_reasons, unapproved_list, isAdmin):
+    try:
+        embed = discord.Embed(
+            title = f'Is Everyone Here',
+            description = f'Members not in meeting channel',
```
Comment (AntWong-TheBadNomads) [inline UI.py:216]:
```suggestion
            description = f'Reports if all relevant full-time members are in the meeting channel',
```
Descriptions should be descriptive

File: Leave/leave_interface.py [hunk 8]
```diff
@@ -154,6 +154,23 @@ def GetReasonOfLeaves(leaves_array):
 
     return leaves_array[0]["reason"]
 
+def GetMissingMembersByRole(members_all, role, voice_channel_members):
```
Comment (AntWong-TheBadNomads) [inline Leave/leave_interface.py:157]: I don't think this function belongs in the leave interface. We don't currently have the most fitting place for this, so I'd put this in utils for now.

Comment (MabroukTBN) [reply Leave/leave_interface.py:157]: GetMissingMemebersByRole uses functions found in *leave_interface* and I did not think it is a good idea to have *utils* import *leave_interface* when *leave_interface* already imports *utils*

File: Leave/leave_interface.py (hunk 3 above)
Comment (MabroukTBN) [reply Leave/leave_interface.py:167]: Given that I will be iterating over them, a dictionary would not be efficient. Instead I'll do any permutations I want within the function before I send them to the embed as a string

File: Leave/leave_interface.py (hunk 2 above)
Comment (MabroukTBN) [reply Leave/leave_interface.py:170]: Will rename to 'missing_members'

File: Leave/leave_interface.py (hunk 4 above)
Comment (MabroukTBN) [reply Leave/leave_interface.py]: This was a mistake. Removed.

File: Leave/leave_interface.py (hunk 5 above)
Comment (MabroukTBN) [reply Leave/leave_interface.py]: Changed variables names to be appropriately descriptive

File: Leave/leave_interface.py (hunk 6 above)
Comment (MabroukTBN) [reply Leave/leave_interface.py]: This was meant to mean both approved and unapproved. Reworded the variable name.

File: Leave/leave_interface.py (hunk 8 above)
Comment (MabroukTBN) [reply Leave/leave_interface.py]: Moved the functionality to IsEveryoneHere in main.py

File: main.py (hunk 1 above)
Comment (MabroukTBN) [reply main.py]: Done

**Pull request merged**